plugins/node/opentelemetry-instrumentation-express/src/express.ts

```diff
@@ -209,6 +209,9 @@ export class ExpressInstrumentation extends InstrumentationBase<
 
         // verify against the config if the layer should be ignored
         if (isLayerIgnored(metadata.name, type, instrumentation._config)) {
+          if (type === ExpressLayerType.MIDDLEWARE) {
```

Comment:
i'm not sure that will do the trick, it only works in the test because we had ignored the `middleware` type which isn't the default. I think we should prefer detecting that the previous handler was mounted the same path as the current one, WDYT ?

Reply:
What do you mean by default? If it's a route handler, then I guess it's fine to have the same path. Afaik the issue only comes to light with middleware layers.

Reply:
your code popup the middleware from the stack when its ignored, i was pointing that it might be not the case. in your tests those middleware are ignored so thats work, but that we'll not be the case for end users

Reply:
Btw this fix is only for ignored layers, if the middleware layers are not ignored it works fine as is.

Reply:
>Btw this fix is only for ignored layers, if the middleware layers are not ignored it works fine as is. 

Thats the thing i don't get, with the following example, the span name is bugged too: 
```ts
// init without ignored layers
const express = new ExpressInstrumentation({})

app.use('/mw', (req, res, next) => {
	  next();
});
	  
app.get('/mw', (req, res) => {
	  res.send('ok');
});

// span name will be `GET /mw/mw` even though it should be `/mw`
```

If so, i think the fix should handle even when the middleware isn't ignored, WDYT ?

Reply:
To clarify, this is with the mainline express instrumentation (not the changes in this PR)?

For example, my demo app:

```javascript
const app = express(); 
app.use('/mw1', function mw1(req, res, next) {
  next();
});
app.use('/mw1', function mw1(req, res, next) {
  next();
});
app.get('/mw1', (req, res) => {
  res.send('ok\n');
}); 
```

With the current main branch:
* `new ExpressInstrumentation()`: root span name = `get /mw1`
* `new ExpressInstrumentation({ ignoreLayersType: [ExpressLayerType.MIDDLEWARE] })`: `get /mw1/mw1/mw1`

With this PR:
* `new ExpressInstrumentation()`: root span name = `get /mw1`
* `new ExpressInstrumentation({ ignoreLayersType: [ExpressLayerType.MIDDLEWARE] })`: `get /mw1`

Or am I missing something here? :smile:

Reply:
No thats definitly me that forgot how the whole middleware stack was working, i was not understanding why with `new ExpressInstrumentation():` root span name = `get /mw1` in the main branch but i looked up the code and i understand know so everything is fine ! Sorry for bothering ahah

Reply:
> No thats definitly me that forgot how the whole middleware stack was working, i was not understanding why with `new ExpressInstrumentation():` root span name = `get /mw1` in the main branch but i looked up the code and i understand know so everything is fine ! Sorry for bothering ahah

No problem, better to get it correct!

I was a bit confused, that perhaps I missed something when testing, so thanks for the double checking :slightly_smiling_face: